rosidl_typesupport_freertps_cpp/CMakeLists.txt

```diff
@@ -40,6 +40,63 @@ ament_export_libraries(${PROJECT_NAME})
 if(AMENT_ENABLE_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
+
+  set(message_files
+    "msg/BoundedArrayBounded.msg"
+    "msg/BoundedArrayStatic.msg"
+    "msg/BoundedArrayUnbounded.msg"
+
+    "msg/Empty.msg"
+    "msg/LargeArray.msg"
+
+    "msg/PrimitiveStaticArrays.msg"
+
+    "msg/PrimitivesBounded.msg"
+    "msg/PrimitivesStatic.msg"
+    "msg/PrimitivesUnbounded.msg"
+
+    "msg/StaticArrayBounded.msg"
+    "msg/StaticArrayStatic.msg"
+    "msg/StaticArrayUnbounded.msg"
+
+    "msg/UnboundedArrayBounded.msg"
+    "msg/UnboundedArrayStatic.msg"
+    "msg/UnboundedArrayUnbounded.msg"
+  )
+
+  include(cmake/register_cpp.cmake)
+  set(rosidl_typesupport_freertps_cpp_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+  rosidl_typesupport_freertps_cpp_extras(
+    "${CMAKE_CURRENT_SOURCE_DIR}/bin/rosidl_typesupport_freertps_cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/rosidl_typesupport_freertps_cpp/__init__.py"
+    "${CMAKE_CURRENT_SOURCE_DIR}/resource"
+  )
+
+  rosidl_generate_interfaces(${PROJECT_NAME}_msgs ${message_files}
+    SKIP_INSTALL
+  )
+
+  if(NOT WIN32)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
+  endif()
+
+  ament_add_gtest(test_serialization_cpp test/test_serialization.cpp)
+  if(TARGET test_serialization_cpp)
+    target_link_libraries(test_serialization_cpp
+      ${PROJECT_NAME}_msgs__rosidl_typesupport_freertps_cpp
+      ${_AMENT_EXPORT_ABSOLUTE_LIBRARIES}
+      ${_AMENT_EXPORT_LIBRARY_TARGETS})
+    add_dependencies(test_serialization_cpp ${PROJECT_NAME} rosidl_generator_cpp)
+  endif()
+  # include the built files directly, instead of their install location
+  include_directories(
+    "${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_freertps_cpp"
+    "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_cpp"
+    include
+    include/${PROJECT_NAME}/impl
+  )
```

Comment:
Since `include_directories` affect all targets this can easily hide a missing include directory which could happen when building without tests. Add the include dirs to the test target explicitly instead.